### PR TITLE
07586 - AddressBookTestingTool Freeze After Genesis Set By Configuration

### LINKED ISSUE #7586: AddressBookTestingTool - Configurable Freeze On Genesis Start
There are a couple JRS test failures due to not freezing before restarting the network in the AddressBookTestingTool tests.  

Solution is to mimic the ConsistencyTestingTool and have a configurable freeze duration added.  

This problem was introduced by turning on PCES that is capturing events created after the last saved state.  Prior restarts would have lost these events.  Now an ISS is occurring.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:

Fixes #7586 

Should be merged concurrently with: 
https://github.com/swirlds/swirlds-platform-regression/pull/3601

### CODE REVIEW

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java [hunk 1]
```diff
@@ -18,20 +18,24 @@
...
- * @param testScenario
- *     The string value of the test scenario being run for validation. This must match an enumerated value in {@link AddressBookTestScenario}.
+ * @param softwareVersion    The integer value of the software version of the AddressBookTestingToolMain SwirldMain
+ *                           application.
+ * @param weightingBehavior  The integer value of the weighting behavior of the AddressBookTestingToolState
+ *                           SwirldState.
+ * @param testScenario       The string value of the test scenario being run for validation. This must match an
+ *                           enumerated value in {@link AddressBookTestScenario}.
+ * @param freezeAfterGenesis if not 0, describes a moment in time, relative to genesis, when a freeze is scheduled
+ *                           {@link com.swirlds.common.config.StateConfig#savedStateDirectory()}.
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java:33]: Do you mean to link `savedStateDirectory()` here? I don't see how it applies

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 2]
```diff
@@ -106,6 +107,24 @@ private static class ClassVersion {
      */
     private long runningSum = 0;
 
+    /**
+     * The number of rounds handled by this app. Is incremented each time
+     * {@link #handleConsensusRound(Round, SwirldDualState)} is called. Note that this may not actually equal the round
+     * number, since we don't call {@link #handleConsensusRound(Round, SwirldDualState)} for rounds with no events.
+     *
+     * <p>
+     * Effects the hash of this node.
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:116]: "Effects" -> "Affects"

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 3]
```diff
@@ -106,6 +107,24 @@ private static class ClassVersion {
...
+     */
+    private long roundsHandled = 0;
+
+    /**
+     * If not zero and we are handling the first round after genesis, configure a freeze this many seconds later.
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:121]: A duration isn't necessarily a number of seconds

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 4]
```diff
@@ -106,6 +107,24 @@ private static class ClassVersion {
...
+     * <p>
+     * Does not affect the hash of this node (although actions may be taken based on this info that DO effect the
```
Comment (alittley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:123]: "...info that DO ~effect~ *affect* the..."

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java:33]: I copied the comment from what was present in the `ConsistencyTestingToolConfig`.   Should the link there be deleted as well?   I don't see how it applies either. :-)

**Review by JeffreyDallas — APPROVED** [review]

**Review by alittley — APPROVED** [review]

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java (hunk 1 above)
Comment (alittley) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolConfig.java]: Lol, so all of the things I found on this PR were just copy-pasted from my own code :D

**Pull request merged**